test/preview/components/readers/test_extractive.py

```diff
@@ -262,7 +262,7 @@ def test_roberta():
 
 @pytest.mark.integration
 def test_matches_hf_pipeline():
-    reader = ExtractiveReader("deepset/tinyroberta-squad2")
+    reader = ExtractiveReader("deepset/tinyroberta-squad2", device="cpu")
```

Comment:
I've seen in the PR description that

> in local environments where GPU is available, this test fails with RuntimeError: Expected all tensors to be on the same device

I think we should open an issue for this? Doesn't sound like expected behavior

Reply:
Here we are comparing HF pipeline with our implementation.

- HF pipeline: [uses the CPU in not specified](https://huggingface.co/docs/transformers/v4.33.2/en/main_classes/pipelines#transformers.QuestionAnsweringPipeline.device)

- Our implementation: https://github.com/deepset-ai/haystack/blob/9a4373bf8e9fe7ebae7d378455f319dc4decd3f0/haystack/preview/components/readers/extractive.py#L40

So, the difference is expected.

Reply:
Hm wait, I understand that we want to run this on CPU, so no problem with the change. I am wondering if running on GPU should raise that error.

Reply:
The error was raised in this line:
https://github.com/deepset-ai/haystack/blob/9a4373bf8e9fe7ebae7d378455f319dc4decd3f0/test/preview/components/readers/test_extractive.py#L270

We are trying to create a HF Pipeline (running on CPU if not specified), passing previously loaded models and tokenizers.

In short, there are no problems if both the Extractive Reader and the HF Pipeline use the same device (CPU or GPU).

In my local experiment, the error pop out because the Extractive Reader was using the GPU, while the HF pipeline tried to use the CPU.

There are different default values for `device`, but it does not seem a bug to me.